common/functions/src/scalars/strings/quote.rs

```diff
@@ -12,33 +12,37 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::io::Write;
+
+use common_datavalues::prelude::DFStringArray;
+
 use super::string2string::String2StringFunction;
 use super::string2string::StringOperator;
 #[derive(Clone, Default)]
-pub struct Quote {
-    buffer: Vec<u8>,
-}
+pub struct Quote {}
 
 impl StringOperator for Quote {
     #[inline]
-    fn apply<'a>(&'a mut self, value: &'a [u8]) -> Option<&'a [u8]> {
-        self.buffer.clear();
-
+    fn apply_with_no_null<'a>(&'a mut self, value: &'a [u8], mut buffer: &mut [u8]) -> usize {
+        let mut len = 0;
         for ch in value {
             match *ch {
-                0 => self.buffer.extend_from_slice(&[b'\\', b'0']),
-                b'\'' => self.buffer.extend_from_slice(&[b'\\', b'\'']),
-                b'\"' => self.buffer.extend_from_slice(&[b'\\', b'\"']),
-                8 => self.buffer.extend_from_slice(&[b'\\', b'b']),
-                b'\n' => self.buffer.extend_from_slice(&[b'\\', b'n']),
-                b'\r' => self.buffer.extend_from_slice(&[b'\\', b'r']),
-                b'\t' => self.buffer.extend_from_slice(&[b'\\', b't']),
-                b'\\' => self.buffer.extend_from_slice(&[b'\\', b'\\']),
-                _ => self.buffer.push(*ch),
-            }
+                0 => len += buffer.write(&[b'\\', b'0']).unwrap_or(0),
```

Comment:
extend_from_slice may be better， this is memory io write, no need to use Trait writer